### PR TITLE
Make wraps more lazy

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/adviser/issues/2014

## This introduces a breaking change

- [x] No

### CODE REVIEW

**Comment by fridex** [comment]
/unhold

ready to be reviewed

File: thoth/adviser/resolver.py [hunk 1]
```diff
@@ -1302,18 +1303,21 @@ def resolve_products(
...
 
         self._init_context()
         with Unit.assigned_context(self.context), self.predictor.assigned_context(self.context):
             for state in self._do_resolve_states(with_devel=with_devel, user_stack_scoring=user_stack_scoring):
-                if report.will_add_product_from_state(state.score):
-                    # An optimization - we do not need to run wraps if the product is not part of the final result.
-                    self._run_wraps(state)
-                    product = Product.from_final_state(context=self.context, state=state)
-                    report.add_product(product)
+                item = ((state.score, heapq_counter), state)
```
Comment (KPostOffice) [inline thoth/adviser/resolver.py:1312]: Can you add a small comment that item[0] is being used only for sorting?

**Review by KPostOffice — COMMENTED** [review]
LGTM just one small comment about adding an inline comment

File: thoth/adviser/resolver.py (hunk 1 above)
Comment (fridex) [reply thoth/adviser/resolver.py:1312]: Adjusted in c6dade99. 👍🏻

**Review by KPostOffice — APPROVED** [review]
Thanks 👍🏻

**Pull request merged**